src/toil/leader.py

```diff
@@ -1107,10 +1107,11 @@ def innerLoop(jobStore, config, batchSystem, toilState, jobBatcher, serviceManag
                     # Remove the job
                     if jobGraph.remainingRetryCount > 0:
                         jobBatcher.issueJob(JobNode.fromJobGraph(jobGraph))
-                        logger.debug("Job: %s is empty, we are scheduling to clean it up", jobGraph.jobStoreID)
+                        logger.debug("Job: %s is empty, we are scheduling to clean it up", jobGraph)
                     else:
                         jobBatcher.processTotallyFailedJob(jobGraph)
-                        logger.warn("Job: %s is empty but completely failed - something is very wrong", jobGraph.jobStoreID)
+                        logger.critical("Job: %s is empty but completely failed - something is "
```

Comment:
I think it makes sense to raise an exception here, this would only execute if an active job has no command, no successors, and is out of retries. I don't think that is ever supposed to happen.